src/main/java/de/retest/recheck/ignore/Filter.java

```diff
@@ -58,6 +75,11 @@ public boolean matches( final Element element ) {
 			return false;
 		}
 
+		@Override
+		public boolean matches( final Element element, final String attributeKey ) {
```

Comment:
Since `matches( Element, AttributeDifference )` defaults to this method, and this method defaults to `matches( Element )`, isn't it sufficient to only overwrite the latter?

Reply:
It is needless to override `matches( Element, String )` but the existing overridden `matches( Element, AttributeDifference )` needs to be retained to avoid `java.lang.NullPointerException` in the below scenario.

```
final AttributeDifference attributeDifference = null;
final Filter filter = Filter.FILTER_NOTHING;
filter.matches( element, attributeDifference );
```

Reply:
Well, I would hope that we can establish with new APIs that no one is allowed to pass `null` as a parameter, unless it is specified.